src/GafferUI/StandardNodule.cpp

```diff
@@ -118,9 +118,30 @@ void StandardNodule::updateDragEndPoint( const Imath::V3f position, const Imath:
  	requestRender();
 }
 
-void StandardNodule::createConnection( Gaffer::Plug *destinationPlug )
+bool StandardNodule::canCreateConnection( const GafferUI::Nodule *endpoint )
+{
+	const Gaffer::Plug *localPlug = plug();
+	const Gaffer::Plug *destinationPlug = endpoint->plug();
+
+	if( localPlug->node() == destinationPlug->node() || localPlug->direction() == destinationPlug->direction() )
+	{
+		return false;
+	}
+
+	if( localPlug->direction() == Gaffer::Plug::Direction::In )
+	{
+		return localPlug->acceptsInput( destinationPlug );
+	}
+	else
+	{
+		return destinationPlug->acceptsInput( localPlug );
+	}
+}
+
+void StandardNodule::createConnection( GafferUI::Nodule *endpoint )
```

Comment:
Now we have `canCreateConnection()`, there's a lot of unnecessary defensive code in this function that will hide bugs by silently doing nothing. It's the responsibility of the caller to verify `canCreateConnection()` before calling `createConnection()`, so we can simplify this down to just the following :

```
if( localPlug->direction() == Gaffer::Plug::Direction::In )
{
    localPlug->setInput( destinationPlug );
}
else
{
 	destinationPlug->setInput( localPlug );
 }
```

Any rogue callers will now be alerted to their error by an exception...